fastlane/lib/fastlane/fast_file.rb

```diff
@@ -313,8 +314,9 @@ def import_from_git(url: nil, branch: 'HEAD', path: 'fastlane/Fastfile', version
               # so we can switch between tags or branches instantly, or else,
               # it would defeat the caching's purpose.
               depth = is_eligible_for_caching ? "" : "--depth 1"
-
-              Actions.sh("git clone #{url.shellescape} #{clone_folder.shellescape} #{depth} --no-checkout #{branch_option}")
+              command = "git clone #{url.shellescape} #{clone_folder.shellescape} #{depth} --no-checkout #{branch_option}"
+              command << " -c http.extraheader='Authorization: Basic #{git_basic_authorization}'" unless git_basic_authorization.nil?
```

Comment:
Even though ["the variable names are case-insensitive"](https://git-scm.com/docs/git-config#:~:text=the%20last%20dot.-,The%20variable%20names%20are%20case%2Dinsensitive,-%2C%20allow%20only%20alphanumeric), I would still use the same casing as in the documentation:

```suggestion
              command << " -c http.extraHeader='Authorization: Basic #{git_basic_authorization}'" unless git_basic_authorization.nil?
```

https://git-scm.com/docs/git-config#Documentation/git-config.txt-httpextraHeader